applications/dairdrop.md

```diff
@@ -0,0 +1,232 @@
+- references to conversations you might have had related to this project with anyone from the Web3 Foundation,
+- previous interface iterations, such as mock-ups and wireframes. -->
+
+## Development Roadmap :nut_and_bolt:
+
+<!-- This section should break the development roadmap down into milestones and deliverables. To assist you in defining it, we have created a document with examples for some grant categories [here](../docs/Support%20Docs/grant_guidelines_per_category.md). Since these will be part of the agreement, it helps to describe *the functionality we should expect in as much detail as possible*, plus how we can verify and test that functionality. Whenever milestones are delivered, we refer to this document to ensure that everything has been delivered as expected.
+
+Below we provide an **example roadmap**. In the descriptions, it should be clear how your project is related to Substrate, Kusama or Polkadot. We *recommend* that teams structure their roadmap as 1 milestone ≈ 1 month. -->
+
+<!-- > :exclamation: If any of your deliverables are based on somebody else's work, make sure you work and publish *under the terms of the license* of the respective project and that you **highlight this fact in your milestone documentation** and in the source code if applicable! **Projects that submit other people's work without proper attribution will be immediately terminated.** -->
+
+### Overview
+
+- **Total Estimated Duration:** 3 months
+- **Full-Time Equivalent (FTE):**  3,5
+<!-- - Average number of full-time employees working on the project throughout its duration (see [Wikipedia](https://en.wikipedia.org/wiki/Full-time_equivalent), e.g. 2 FTE) -->
+- **Total Costs:** 20,000
+- **DOT %:** 50%
+
+### Milestone 1 — On-chain pallets and Ink contracts, airdrop indexer, API services, and more.
+
+- **Estimated duration:** 1.5 month
+- **FTE:**  2,0
+- **Costs:** 10,000 USD
+
+> :exclamation: **The default deliverables 0a-0d below are mandatory for all milestones**, and deliverable 0e at least for the last one.
+
+| Number | Deliverable | Specification |
+| -----: | ----------- | ------------- |
+| **0a.** | License | Apache 2.0  |
+| **0b.** | Documentation | Create detailed documentation for each module. |
+| **0c.** | Testing and Testing Guide | Over 70% code test coverage. |
+| **0d.** | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 1. | Substrate module: airdorp | An airdrop module that provides all on-chain functionalities related to airdrops |
+| 2. | Smart contracts:  airdorp-ink| Implementation of the airdrop module using Ink. |
+| 3. | indexer| An on-chain airdrop activity indexer that monitors on-chain airdrop status in real time.|
+| 3. | backend | Provide all APIs needed for the app, including calculating the MD5 hash of the airdrop file, the Merkle root of the airdrop, Merkle proofs, and airdrop status. |
```

Comment:
what is the backend? I suppose the substrate pallet and the ink implementation serve directly as the backend.

Reply:
> what is the backend? I suppose the substrate pallet and the ink implementation serve directly as the backend.

On-chain storage primarily contains the final state and completed Merkle proof verification. We need to provide an off-chain backend that allows users to obtain their Merkle proofs, submit transactions to claim airdrops, and access more information about the airdrop, such as progress, the number of participants, and which users have not yet participated. Additionally, anyone should be able to claim airdrops for others.

Reply:
isn't this part of the web-service?

Reply:
> isn't this part of the web-service?

Yes, considering the PC app, I developed it as part of the public section. The web server simply calls the APIs it needs from there.